py/selenium/webdriver/remote/websocket_connection.py

```diff
@@ -0,0 +1,107 @@
+import json
+import logging
+from ssl import CERT_NONE
+from time import sleep
+from threading import Thread
+from websocket import WebSocketApp
+
+logger = logging.getLogger("websocket")
+
+class WebSocketConnection:
+    _response_wait_timeout = 30
+    _response_wait_interval = 0.1
+
+    _max_log_message_size = 9999
+
+    def __init__(self, url):
+        self.session_id = None
+        self.url = url
+
+        self._id = 0
+        self._callbacks = {}
+        self._messages = {}
+        self._started = False
+
+        self._start_ws()
+        self._wait_until(lambda: self._started)
+
+    def close(self):
+        self._ws_thread.join()
+        self._ws.close()
+        self._started = False
+        self._ws = None
+
+    def execute(self, command):
```

Comment:
A general comment that applies to this whole PR - do we want to have typing properly implemented? I'm not familiar with the Python typing system so I did what was the easiest for me. I can re-iterate and add types if that's preferred.

Reply:
Types are already available in CDP, the question is should I add types to the `WebSocketConnection` class.

Reply:
I don't think this should be the priority right now. Focus on getting the hard parts implemented, and we can get help from others on the typing later.